src/Modal/index.js

```diff
@@ -318,14 +328,30 @@ class Modal extends React.PureComponent<Props, State> {
     }
   };
 
+  decideModalSection = () => {
+    const children = React.Children.toArray(this.props.children);
+    const hasModalSection =
+      children.map(child => child.type.displayName).indexOf(ModalSection.displayName) !== -1;
```

Comment:
I am not sure if `.displayName` is right, in the other files we are using just `.name`